misk/src/test/kotlin/misk/web/InvalidActionsTest.kt

```diff
@@ -50,4 +51,52 @@ class InvalidActionsTest {
     @ResponseContentType("application/json")
     fun hello() = "hello"
   }
+
+  @Test fun failGrpcWithHttp2Disabled() {
+    val exception = assertThrows<ProvisionException>("Should throw an exception") {
+      Guice.createInjector(GrpcActionsModule()).getInstance(ServiceManager::class.java)
+        .startAsync().awaitHealthy()
+    }
+    assertThat(exception.message).contains(
+      "HTTP/2 must be enabled if any gRPC actions are bound."
+    )
+  }
+
+  class GrpcActionsModule : KAbstractModule() {
+    override fun configure() {
+      install(
+        WebServerTestingModule(
+          webConfig = WebServerTestingModule.TESTING_WEB_CONFIG.copy(
+            http2 = false
+          )
+        )
+      )
+      install(MiskTestingServiceModule())
+      install(WebActionModule.create<HelloRpcAction>())
+    }
+  }
+
+  @Singleton
+  class HelloRpcAction @Inject constructor() : WebAction, GreeterSayHello {
+    var failNextRequest = false
+    var sleep: Duration = Duration.ofMillis(0)
+
+    override fun sayHello(request: HelloRequest): HelloReply {
+      Thread.sleep(sleep.toMillis())
```

Comment:
This is really needed?

Reply:
No this was copy-paste. Can simplify this a bit more.